python/ray/data/_internal/execution/interfaces/op_runtime_metrics.py

```diff
@@ -201,13 +203,27 @@ def output_buffer_bytes(self) -> int:
         """Size in bytes of output blocks that are not taken by the downstream yet."""
         return self.bytes_outputs_generated - self.bytes_outputs_taken
 
+    @property
+    def obj_store_mem_cur(self) -> int:
+        return self.obj_store_mem_inputs + self.obj_store_mem_outputs
+
+    @property
+    def obj_store_mem_cur_upper_bound(self) -> int:
+        if self.average_bytes_outputs_per_task is not None:
+            return self.obj_store_mem_inputs + max(
+                self.obj_store_mem_outputs,
+                self.num_tasks_running * self.average_bytes_outputs_per_task,
```

Comment:
I guess it should be something like `self.obj_store_mem_outputs + sum(self.average_bytes_outputs_per_task - task.cur_output_size for task in running_tasks)`.
But if we also consider streaming backpressure. then the formula will be `self.obj_store_mem_outputs + self.num_tasks_running * max_pending_blocks_at_streaming_gen * avg_block_size`

Reply:
> `obj_store_mem_outputs` also includes the output data from finished tasks, but has not been taken yet

We take data immediately after a task is finished, so this shouldn't be an issue, right? They happen sequentially in `process_completed_tasks`?

That said, I agree `self.obj_store_mem_outputs + sum(self.average_bytes_outputs_per_task - task.cur_output_size for task in running_tasks)` is more accurate. I'll update to use.

> self.obj_store_mem_outputs + self.num_tasks_running * max_pending_blocks_at_streaming_gen * avg_block_size

If `max_pending_blocks_at_streaming_gen ` is large and each task produces only one block, this would be an overestimation?

Reply:
`self.obj_store_mem_outputs + sum(min(self.average_outputs_per_task, max_pending_blocks_at_streaming_gen) * avg_block_size - task.cur_output_size for task in running_tasks)` might work. WDYT?

Reply:
Discussed offline -- we'll go with `self.num_tasks_running * max_pending_blocks_at_streaming_gen * avg_block_size` to estimate in-flight data